src/redux/actions/genes/loadDownsampledGeneExpression.js

```diff
@@ -0,0 +1,138 @@
+import _ from 'lodash';
+import { SparseMatrix } from 'mathjs';
+
+import {
+  DOWNSAMPLED_GENES_LOADING,
+  DOWNSAMPLED_GENES_LOADED,
+  DOWNSAMPLED_GENES_ERROR,
+} from 'redux/actionTypes/genes';
+
+import fetchWork from 'utils/work/fetchWork';
+import getTimeoutForWorkerTask from 'utils/getTimeoutForWorkerTask';
+
+// Debounce so that we only fetch once the settings are done being set up
+const loadDownsampledGeneExpressionDebounced = _.debounce(
+  async (
+    experimentId,
+    genes,
+    componentUuid,
+    dispatch,
+    getState,
+  ) => {
+    const state = getState();
+
+    const {
+      groupedTracks,
+      selectedCellSet,
+      selectedPoints,
+    } = state.componentConfig[componentUuid]?.config;
+
+    const hiddenCellSets = Array.from(state.cellSets.hidden);
+
+    const downsampleSettings = {
+      selectedCellSet,
+      groupedTracks,
+      selectedPoints,
+      hiddenCellSets,
+    };
+
+    if (genes.length === 0) {
+      dispatch({
+        type: DOWNSAMPLED_GENES_LOADED,
+        payload: {
+          componentUuid,
+          genes,
+        },
+      });
+
+      return;
+    }
+
+    const body = {
+      name: 'GeneExpression',
+      genes,
+      downsampled: true,
+      downsampleSettings,
+    };
+
+    const timeout = getTimeoutForWorkerTask(getState(), 'GeneExpression');
+
+    try {
+      let requestETag;
+
+      const {
+        orderedGeneNames,
+        rawExpression: rawExpressionJson,
+        truncatedExpression: truncatedExpressionJson,
+        zScore: zScoreJson,
+        stats,
+        cellOrder,
+      } = await fetchWork(
```

Comment:
Originally the `loadGeneExpression` action creator could check a `downsampled` bool and decide what to do based on that. 

However, I think `loadGeneExpression` and `loadDownsampledGeneExpression` have become different enough to warrant splitting them up (otherwise, it was becoming a mess of ifs checking `downsampled` all the time). 

I do think they might still share behavior in a lot of pieces of code that could be abstracted. Let me know if there is anything you think I might have missed here.

Reply:
Can you reuse the fetchWork part? Specifically the try/catch part. It seems quite similar. Can we have a function separately doing the operations needed for the downsampledGenes which calls the fetch, which can be used in both cases?

Reply:
Y, that seems very similar. I'll take a look